src/main/modules/ardoq/RequestProcessor.ts

```diff
@@ -0,0 +1,42 @@
+import { ArdoqClient } from './ArdoqClient';
+import { ArdoqComponentCreatedResponse } from './ArdoqComponentCreatedResponse';
+import { Dependency } from './Dependency';
+
+import express from 'express';
+
+export class RequestProcessor {
+  client: ArdoqClient;
+
+  constructor(client: ArdoqClient) {
+    this.client = client;
+  }
+
+  public async processRequest(res: express.Response, deps: Map<string, Dependency>): Promise<void> {
+    try {
+      const depUpdate: Promise<ArdoqComponentCreatedResponse>[] = [];
+      deps.forEach((d: Dependency) => {
```

Comment:
personal preference but when doing a for loop I normally go for `for (const d of deps)` rather than `.forEach` - no practical difference though

Reply:
in fact, you want a `map` here:
```
const depUpdate = deps.map(d => this.client.updateDeb(d));
```

Reply:
`TS2339: Property 'map' does not exist on type 'Map '.` :S